src/main/java/com/google/devtools/build/lib/remote/AbstractRemoteActionCache.java

```diff
@@ -68,6 +68,10 @@
     ((SettableFuture<byte[]>) EMPTY_BYTES).set(new byte[0]);
   }
 
+  public static boolean causedByCacheMiss(IOException t) {
+    return t.getCause() instanceof CacheNotFoundException;
```

Comment:
Maybe add a t instanceOf CacheNotFoundException clause just in case?

Reply:
Switched it to RetryException per the only viable typed calls. Until RetryException isn't a part of the interface, it will continue to be the only way it can contain CNFE